src/holofuel/pages/TransactionHistory/TransactionHistory.js

```diff
@@ -27,12 +26,55 @@ function useCancel () {
   })
 }
 
+function useFetchCounterparties () {
+  const { data: { holofuelCompletedTransactions = [] } = {} } = useQuery(HolofuelCompletedTransactionsQuery)
+  const { data: { holofuelWaitingTransactions = [] } = {} } = useQuery(HolofuelWaitingTransactionsQuery)
+
+  const { loading, error, data: { holofuelHistoryCounterparties } = {}, client } = useQuery(HolofuelHistoryCounterpartiesQuery)
+
+  if (holofuelHistoryCounterparties) {
+    const filterTransactionsByAgentId = (agent, txListType) => txListType.filter(transaction => transaction.counterparty.id === agent.id)
+    const updateTxListCounterparties = (txListType, counterpartyList) => counterpartyList.map(agent => {
+      const matchingTx = filterTransactionsByAgentId(agent, txListType)
+      return matchingTx.map(transaction => { Object.assign(transaction.counterparty, agent); return transaction })
+    })
+
+    // Cache Write/Update for HolofuelCompletedTransactionsQuery
+    const newCompletedTxList = flatten(updateTxListCounterparties(holofuelCompletedTransactions, holofuelHistoryCounterparties))
+    client.writeQuery({
+      query: HolofuelCompletedTransactionsQuery,
+      data: {
+        holofuelCompletedTransactions: newCompletedTxList
+      }
+    })
+
+    // Cache Write/Update for HolofuelWaitingTransactionsQuery
+    const newWaitingTxList = flatten(updateTxListCounterparties(holofuelWaitingTransactions, holofuelHistoryCounterparties))
+    client.writeQuery({
+      query: HolofuelWaitingTransactionsQuery,
+      data: {
+        holofuelWaitingTransactions: newWaitingTxList
+      }
+    })
+  }
+
+  let response
+  if (loading) response = { loading: true }
+  else if (error) response = { error: `Error: ${error}` }
+  else response = holofuelHistoryCounterparties
+  return response
+}
+
 // Display - Functional Components with Hooks :
 export default function TransactionsHistory () {
   const { data: { holofuelCompletedTransactions: completedTransactions = [] } = {} } = useQuery(HolofuelCompletedTransactionsQuery)
   const { data: { holofuelWaitingTransactions: pendingTransactions = [] } = {} } = useQuery(HolofuelWaitingTransactionsQuery)
 
   const cancelTransaction = useCancel()
+
+  useFetchCounterparties()
+  // const counterpartyList = useFetchCounterparties()
```

Comment:
remove commented code and move this up to the top with the other queries as in Inbox